### PR TITLE
move Clickhouse HTTP handler to its own port.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

1. default 8124, (clickhouse  use 8123)
2. `:8000/clickhouse` not remote in this pr.

## Changelog

- New Feature

## Related Issues

Fixes https://github.com/datafuselabs/databend/issues/5728

### CODE REVIEW

File: docker/databend-query-docker.toml [hunk 1]
```diff
@@ -23,6 +23,10 @@ mysql_handler_port = 3307
 clickhouse_handler_host = "0.0.0.0"
 clickhouse_handler_port = 9001
 
+# Databend Query ClickHouse Handler.
+clickhouse_http_handler_host = "0.0.0.0"
```
Comment (Xuanwo) [inline docker/databend-query-docker.toml:27]: How about providing a `clickhouse_http_handler_addr` instead?

Comment (youngsofun) [reply docker/databend-query-docker.toml:27]: SGTM, but not quite sure.

File: docker/databend-query-docker.toml [hunk 2]
```diff
@@ -23,6 +23,10 @@ mysql_handler_port = 3307
 clickhouse_handler_host = "0.0.0.0"
 clickhouse_handler_port = 9001
 
+# Databend Query ClickHouse Handler.
```
Comment (youngsofun) [inline docker/databend-query-docker.toml:26]: wrong comment

**Review by ZhiHanZ — APPROVED** [review]
nice work!

**Comment by youngsofun** [comment]
failure tests seem not relevant:

```
Having 7 errors! 154 tests passed.                     0 tests skipped.
The failure tests:
    /workspace/tests/suites/0_stateless/02_function/02_0004_function_name_display.sql
    /workspace/tests/suites/0_stateless/03_dml/03_0004_select_order_by.sql
    /workspace/tests/suites/0_stateless/03_dml/03_0005_select_filter.sql
    /workspace/tests/suites/0_stateless/08_optimizer/08_0000_optimizer.sql
    /workspace/tests/suites/0_stateless/08_optimizer/08_0001_optimizer_statistics_exact.sql
    /workspace/tests/suites/0_stateless/09_fuse_engine/09_0009_remote_explain.sql
    /workspace/tests/suites/0_stateless/09_fuse_engine/09_0012_pushdown_limit.sql
```

File: docs/doc/10-deploy/00-local.md [hunk 3]
```diff
@@ -140,6 +140,10 @@ mysql_handler_port = 3307
 clickhouse_handler_host = "127.0.0.1"
 clickhouse_handler_port = 9001
 
+# Query ClickHouse HTTP Handler.
+clickhouse_http_handler_host = "127.0.0.1"
+clickhouse_http_handler_port = 9001
```
Comment (BohuTANG) [inline docs/doc/10-deploy/00-local.md:145]: Same as `clickhouse_handler_port`?

**Comment by BohuTANG** [comment]
@mergify update

**Comment by BohuTANG** [comment]
It seems that this PR break some tests.

**Comment by youngsofun** [comment]
> It seems that this PR break some tests.

my typo: dup ports in the config of servers for cluster test

**Comment by youngsofun** [comment]
@BohuTANG fixed

**Comment by sundy-li** [comment]
@youngsofun  Prefer to make  `clickhouse_http_handler_port` default to 8123, some drivers will default to 8123 if it's not specified.

**Comment by youngsofun** [comment]
@sundy-li  user use this is likely to migrate from clickhouse, so he may have clickhouse running on his machine， in this situation， databend will fail to start.  or maybe user wants to start both and do some compare?

is this why we use 3307 for MySQL?  cc @BohuTANG

**Pull request merged**